// @hubspot/api-client/lib/codegen/crm/lists/models/PublicWeekReference.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicWeekReferenceDayOfWeekEnum = exports.PublicWeekReferenceReferenceTypeEnum = exports.PublicWeekReference = void 0;
class PublicWeekReference {
    static getAttributeTypeMap() {
        return PublicWeekReference.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicWeekReference = PublicWeekReference;
PublicWeekReference.discriminator = undefined;
PublicWeekReference.attributeTypeMap = [
    {
        "name": "referenceType",
        "baseName": "referenceType",
        "type": "PublicWeekReferenceReferenceTypeEnum",
        "format": ""
    },
    {
        "name": "hour",
        "baseName": "hour",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "minute",
        "baseName": "minute",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "second",
        "baseName": "second",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "millisecond",
        "baseName": "millisecond",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "dayOfWeek",
        "baseName": "dayOfWeek",
        "type": "PublicWeekReferenceDayOfWeekEnum",
        "format": ""
    }
];
var PublicWeekReferenceReferenceTypeEnum;
(function (PublicWeekReferenceReferenceTypeEnum) {
    PublicWeekReferenceReferenceTypeEnum["Week"] = "WEEK";
})(PublicWeekReferenceReferenceTypeEnum = exports.PublicWeekReferenceReferenceTypeEnum || (exports.PublicWeekReferenceReferenceTypeEnum = {}));
var PublicWeekReferenceDayOfWeekEnum;
(function (PublicWeekReferenceDayOfWeekEnum) {
    PublicWeekReferenceDayOfWeekEnum["Monday"] = "MONDAY";
    PublicWeekReferenceDayOfWeekEnum["Tuesday"] = "TUESDAY";
    PublicWeekReferenceDayOfWeekEnum["Wednesday"] = "WEDNESDAY";
    PublicWeekReferenceDayOfWeekEnum["Thursday"] = "THURSDAY";
    PublicWeekReferenceDayOfWeekEnum["Friday"] = "FRIDAY";
    PublicWeekReferenceDayOfWeekEnum["Saturday"] = "SATURDAY";
    PublicWeekReferenceDayOfWeekEnum["Sunday"] = "SUNDAY";
})(PublicWeekReferenceDayOfWeekEnum = exports.PublicWeekReferenceDayOfWeekEnum || (exports.PublicWeekReferenceDayOfWeekEnum = {}));
