// @hubspot/api-client/lib/codegen/crm/lists/models/PublicCalendarDatePropertyOperation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicCalendarDatePropertyOperationFiscalYearStartEnum = exports.PublicCalendarDatePropertyOperationOperationTypeEnum = exports.PublicCalendarDatePropertyOperation = void 0;
class PublicCalendarDatePropertyOperation {
    static getAttributeTypeMap() {
        return PublicCalendarDatePropertyOperation.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PublicCalendarDatePropertyOperation = PublicCalendarDatePropertyOperation;
PublicCalendarDatePropertyOperation.discriminator = undefined;
PublicCalendarDatePropertyOperation.attributeTypeMap = [
    {
        "name": "operationType",
        "baseName": "operationType",
        "type": "PublicCalendarDatePropertyOperationOperationTypeEnum",
        "format": ""
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "string",
        "format": ""
    },
    {
        "name": "includeObjectsWithNoValueSet",
        "baseName": "includeObjectsWithNoValueSet",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "timeUnit",
        "baseName": "timeUnit",
        "type": "string",
        "format": ""
    },
    {
        "name": "fiscalYearStart",
        "baseName": "fiscalYearStart",
        "type": "PublicCalendarDatePropertyOperationFiscalYearStartEnum",
        "format": ""
    },
    {
        "name": "useFiscalYear",
        "baseName": "useFiscalYear",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "timeUnitCount",
        "baseName": "timeUnitCount",
        "type": "number",
        "format": "int32"
    }
];
var PublicCalendarDatePropertyOperationOperationTypeEnum;
(function (PublicCalendarDatePropertyOperationOperationTypeEnum) {
    PublicCalendarDatePropertyOperationOperationTypeEnum["CalendarDate"] = "CALENDAR_DATE";
})(PublicCalendarDatePropertyOperationOperationTypeEnum = exports.PublicCalendarDatePropertyOperationOperationTypeEnum || (exports.PublicCalendarDatePropertyOperationOperationTypeEnum = {}));
var PublicCalendarDatePropertyOperationFiscalYearStartEnum;
(function (PublicCalendarDatePropertyOperationFiscalYearStartEnum) {
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["January"] = "JANUARY";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["February"] = "FEBRUARY";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["March"] = "MARCH";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["April"] = "APRIL";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["May"] = "MAY";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["June"] = "JUNE";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["July"] = "JULY";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["August"] = "AUGUST";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["September"] = "SEPTEMBER";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["October"] = "OCTOBER";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["November"] = "NOVEMBER";
    PublicCalendarDatePropertyOperationFiscalYearStartEnum["December"] = "DECEMBER";
})(PublicCalendarDatePropertyOperationFiscalYearStartEnum = exports.PublicCalendarDatePropertyOperationFiscalYearStartEnum || (exports.PublicCalendarDatePropertyOperationFiscalYearStartEnum = {}));
