// @hubspot/api-client/lib/codegen/crm/extensions/accounting/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectUserAccountsApi = exports.ObjectSyncApi = exports.ObjectSettingsApi = exports.ObjectInvoiceApi = exports.ObjectCallbacksApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectCallbacksApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableCallbacksApi(configuration, requestFactory, responseProcessor);
    }
    createCustomerWithHttpInfo(param, options) {
        return this.api.createCustomerWithHttpInfo(param.requestId, param.resultIdAccountingResponse, options).toPromise();
    }
    createCustomer(param, options) {
        return this.api.createCustomer(param.requestId, param.resultIdAccountingResponse, options).toPromise();
    }
    createExchangeRateWithHttpInfo(param, options) {
        return this.api.createExchangeRateWithHttpInfo(param.requestId, param.exchangeRateResponse, options).toPromise();
    }
    createExchangeRate(param, options) {
        return this.api.createExchangeRate(param.requestId, param.exchangeRateResponse, options).toPromise();
    }
    createInvoiceWithHttpInfo(param, options) {
        return this.api.createInvoiceWithHttpInfo(param.requestId, param.resultIdAccountingResponse, options).toPromise();
    }
    createInvoice(param, options) {
        return this.api.createInvoice(param.requestId, param.resultIdAccountingResponse, options).toPromise();
    }
    createTermWithHttpInfo(param, options) {
        return this.api.createTermWithHttpInfo(param.requestId, param.termsResponse, options).toPromise();
    }
    createTerm(param, options) {
        return this.api.createTerm(param.requestId, param.termsResponse, options).toPromise();
    }
    doCustomerSearchWithHttpInfo(param, options) {
        return this.api.doCustomerSearchWithHttpInfo(param.requestId, param.customerSearchResponseExternal, options).toPromise();
    }
    doCustomerSearch(param, options) {
        return this.api.doCustomerSearch(param.requestId, param.customerSearchResponseExternal, options).toPromise();
    }
    doInvoiceSearchWithHttpInfo(param, options) {
        return this.api.doInvoiceSearchWithHttpInfo(param.requestId, param.invoiceSearchResponse, options).toPromise();
    }
    doInvoiceSearch(param, options) {
        return this.api.doInvoiceSearch(param.requestId, param.invoiceSearchResponse, options).toPromise();
    }
    doProductSearchWithHttpInfo(param, options) {
        return this.api.doProductSearchWithHttpInfo(param.requestId, param.productSearchResponse, options).toPromise();
    }
    doProductSearch(param, options) {
        return this.api.doProductSearch(param.requestId, param.productSearchResponse, options).toPromise();
    }
    doTaxSearchWithHttpInfo(param, options) {
        return this.api.doTaxSearchWithHttpInfo(param.requestId, param.taxSearchResponse, options).toPromise();
    }
    doTaxSearch(param, options) {
        return this.api.doTaxSearch(param.requestId, param.taxSearchResponse, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.requestId, param.invoicesResponseExternal, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.requestId, param.invoicesResponseExternal, options).toPromise();
    }
    invoicePdfWithHttpInfo(param, options) {
        return this.api.invoicePdfWithHttpInfo(param.requestId, param.invoicePdfResponse, options).toPromise();
    }
    invoicePdf(param, options) {
        return this.api.invoicePdf(param.requestId, param.invoicePdfResponse, options).toPromise();
    }
}
exports.ObjectCallbacksApi = ObjectCallbacksApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectInvoiceApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableInvoiceApi(configuration, requestFactory, responseProcessor);
    }
    createPaymentWithHttpInfo(param, options) {
        return this.api.createPaymentWithHttpInfo(param.invoiceId, param.invoiceCreatePaymentRequest, param.accountId, options).toPromise();
    }
    createPayment(param, options) {
        return this.api.createPayment(param.invoiceId, param.invoiceCreatePaymentRequest, param.accountId, options).toPromise();
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.invoiceId, param.accountId, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.invoiceId, param.accountId, options).toPromise();
    }
    updateWithHttpInfo(param, options) {
        return this.api.updateWithHttpInfo(param.invoiceId, param.accountId, param.invoiceUpdateRequest, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.invoiceId, param.accountId, param.invoiceUpdateRequest, options).toPromise();
    }
}
exports.ObjectInvoiceApi = ObjectInvoiceApi;
const ObservableAPI_3 = require("./ObservableAPI");
class ObjectSettingsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableSettingsApi(configuration, requestFactory, responseProcessor);
    }
    getByIdWithHttpInfo(param, options) {
        return this.api.getByIdWithHttpInfo(param.appId, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.appId, options).toPromise();
    }
    replaceWithHttpInfo(param, options) {
        return this.api.replaceWithHttpInfo(param.appId, param.accountingAppSettings, options).toPromise();
    }
    replace(param, options) {
        return this.api.replace(param.appId, param.accountingAppSettings, options).toPromise();
    }
}
exports.ObjectSettingsApi = ObjectSettingsApi;
const ObservableAPI_4 = require("./ObservableAPI");
class ObjectSyncApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_4.ObservableSyncApi(configuration, requestFactory, responseProcessor);
    }
    createContactWithHttpInfo(param, options) {
        return this.api.createContactWithHttpInfo(param.appId, param.syncContactsRequest, options).toPromise();
    }
    createContact(param, options) {
        return this.api.createContact(param.appId, param.syncContactsRequest, options).toPromise();
    }
    createProductWithHttpInfo(param, options) {
        return this.api.createProductWithHttpInfo(param.appId, param.syncProductsRequest, options).toPromise();
    }
    createProduct(param, options) {
        return this.api.createProduct(param.appId, param.syncProductsRequest, options).toPromise();
    }
}
exports.ObjectSyncApi = ObjectSyncApi;
const ObservableAPI_5 = require("./ObservableAPI");
class ObjectUserAccountsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_5.ObservableUserAccountsApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(param, options) {
        return this.api.archiveWithHttpInfo(param.accountId, options).toPromise();
    }
    archive(param, options) {
        return this.api.archive(param.accountId, options).toPromise();
    }
    replaceWithHttpInfo(param, options) {
        return this.api.replaceWithHttpInfo(param.createUserAccountRequestExternal, options).toPromise();
    }
    replace(param, options) {
        return this.api.replace(param.createUserAccountRequestExternal, options).toPromise();
    }
}
exports.ObjectUserAccountsApi = ObjectUserAccountsApi;
