// @hubspot/api-client/lib/codegen/crm/objects/feedback_submissions/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseSearchApi = exports.PromisePublicObjectApi = exports.PromiseGDPRApi = exports.PromiseBatchApi = exports.PromiseBasicApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseBasicApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableBasicApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(feedbackSubmissionId, _options) {
        const result = this.api.archiveWithHttpInfo(feedbackSubmissionId, _options);
        return result.toPromise();
    }
    archive(feedbackSubmissionId, _options) {
        const result = this.api.archive(feedbackSubmissionId, _options);
        return result.toPromise();
    }
    createWithHttpInfo(simplePublicObjectInputForCreate, _options) {
        const result = this.api.createWithHttpInfo(simplePublicObjectInputForCreate, _options);
        return result.toPromise();
    }
    create(simplePublicObjectInputForCreate, _options) {
        const result = this.api.create(simplePublicObjectInputForCreate, _options);
        return result.toPromise();
    }
    getByIdWithHttpInfo(feedbackSubmissionId, properties, propertiesWithHistory, associations, archived, idProperty, _options) {
        const result = this.api.getByIdWithHttpInfo(feedbackSubmissionId, properties, propertiesWithHistory, associations, archived, idProperty, _options);
        return result.toPromise();
    }
    getById(feedbackSubmissionId, properties, propertiesWithHistory, associations, archived, idProperty, _options) {
        const result = this.api.getById(feedbackSubmissionId, properties, propertiesWithHistory, associations, archived, idProperty, _options);
        return result.toPromise();
    }
    getPageWithHttpInfo(limit, after, properties, propertiesWithHistory, associations, archived, _options) {
        const result = this.api.getPageWithHttpInfo(limit, after, properties, propertiesWithHistory, associations, archived, _options);
        return result.toPromise();
    }
    getPage(limit, after, properties, propertiesWithHistory, associations, archived, _options) {
        const result = this.api.getPage(limit, after, properties, propertiesWithHistory, associations, archived, _options);
        return result.toPromise();
    }
    updateWithHttpInfo(feedbackSubmissionId, simplePublicObjectInput, idProperty, _options) {
        const result = this.api.updateWithHttpInfo(feedbackSubmissionId, simplePublicObjectInput, idProperty, _options);
        return result.toPromise();
    }
    update(feedbackSubmissionId, simplePublicObjectInput, idProperty, _options) {
        const result = this.api.update(feedbackSubmissionId, simplePublicObjectInput, idProperty, _options);
        return result.toPromise();
    }
}
exports.PromiseBasicApi = PromiseBasicApi;
const ObservableAPI_2 = require("./ObservableAPI");
class PromiseBatchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableBatchApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(batchInputSimplePublicObjectId, _options) {
        const result = this.api.archiveWithHttpInfo(batchInputSimplePublicObjectId, _options);
        return result.toPromise();
    }
    archive(batchInputSimplePublicObjectId, _options) {
        const result = this.api.archive(batchInputSimplePublicObjectId, _options);
        return result.toPromise();
    }
    createWithHttpInfo(batchInputSimplePublicObjectInputForCreate, _options) {
        const result = this.api.createWithHttpInfo(batchInputSimplePublicObjectInputForCreate, _options);
        return result.toPromise();
    }
    create(batchInputSimplePublicObjectInputForCreate, _options) {
        const result = this.api.create(batchInputSimplePublicObjectInputForCreate, _options);
        return result.toPromise();
    }
    readWithHttpInfo(batchReadInputSimplePublicObjectId, archived, _options) {
        const result = this.api.readWithHttpInfo(batchReadInputSimplePublicObjectId, archived, _options);
        return result.toPromise();
    }
    read(batchReadInputSimplePublicObjectId, archived, _options) {
        const result = this.api.read(batchReadInputSimplePublicObjectId, archived, _options);
        return result.toPromise();
    }
    updateWithHttpInfo(batchInputSimplePublicObjectBatchInput, _options) {
        const result = this.api.updateWithHttpInfo(batchInputSimplePublicObjectBatchInput, _options);
        return result.toPromise();
    }
    update(batchInputSimplePublicObjectBatchInput, _options) {
        const result = this.api.update(batchInputSimplePublicObjectBatchInput, _options);
        return result.toPromise();
    }
}
exports.PromiseBatchApi = PromiseBatchApi;
const ObservableAPI_3 = require("./ObservableAPI");
class PromiseGDPRApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableGDPRApi(configuration, requestFactory, responseProcessor);
    }
    purgeWithHttpInfo(publicGdprDeleteInput, _options) {
        const result = this.api.purgeWithHttpInfo(publicGdprDeleteInput, _options);
        return result.toPromise();
    }
    purge(publicGdprDeleteInput, _options) {
        const result = this.api.purge(publicGdprDeleteInput, _options);
        return result.toPromise();
    }
}
exports.PromiseGDPRApi = PromiseGDPRApi;
const ObservableAPI_4 = require("./ObservableAPI");
class PromisePublicObjectApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_4.ObservablePublicObjectApi(configuration, requestFactory, responseProcessor);
    }
    mergeWithHttpInfo(publicMergeInput, _options) {
        const result = this.api.mergeWithHttpInfo(publicMergeInput, _options);
        return result.toPromise();
    }
    merge(publicMergeInput, _options) {
        const result = this.api.merge(publicMergeInput, _options);
        return result.toPromise();
    }
}
exports.PromisePublicObjectApi = PromisePublicObjectApi;
const ObservableAPI_5 = require("./ObservableAPI");
class PromiseSearchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_5.ObservableSearchApi(configuration, requestFactory, responseProcessor);
    }
    doSearchWithHttpInfo(publicObjectSearchRequest, _options) {
        const result = this.api.doSearchWithHttpInfo(publicObjectSearchRequest, _options);
        return result.toPromise();
    }
    doSearch(publicObjectSearchRequest, _options) {
        const result = this.api.doSearch(publicObjectSearchRequest, _options);
        return result.toPromise();
    }
}
exports.PromiseSearchApi = PromiseSearchApi;
