// @hubspot/api-client/lib/codegen/marketing/events/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseSubscriberStateChangesApi = exports.PromiseSettingsApi = exports.PromiseSearchApi = exports.PromiseMarketingEventsExternalApi = exports.PromiseBatchApi = exports.PromiseBasicApi = exports.PromiseAttendanceSubscriberStateChangesApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseAttendanceSubscriberStateChangesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableAttendanceSubscriberStateChangesApi(configuration, requestFactory, responseProcessor);
    }
    createWithHttpInfo(externalEventId, subscriberState, batchInputMarketingEventSubscriber, externalAccountId, _options) {
        const result = this.api.createWithHttpInfo(externalEventId, subscriberState, batchInputMarketingEventSubscriber, externalAccountId, _options);
        return result.toPromise();
    }
    create(externalEventId, subscriberState, batchInputMarketingEventSubscriber, externalAccountId, _options) {
        const result = this.api.create(externalEventId, subscriberState, batchInputMarketingEventSubscriber, externalAccountId, _options);
        return result.toPromise();
    }
    createByEmailWithHttpInfo(externalEventId, subscriberState, batchInputMarketingEventEmailSubscriber, externalAccountId, _options) {
        const result = this.api.createByEmailWithHttpInfo(externalEventId, subscriberState, batchInputMarketingEventEmailSubscriber, externalAccountId, _options);
        return result.toPromise();
    }
    createByEmail(externalEventId, subscriberState, batchInputMarketingEventEmailSubscriber, externalAccountId, _options) {
        const result = this.api.createByEmail(externalEventId, subscriberState, batchInputMarketingEventEmailSubscriber, externalAccountId, _options);
        return result.toPromise();
    }
}
exports.PromiseAttendanceSubscriberStateChangesApi = PromiseAttendanceSubscriberStateChangesApi;
const ObservableAPI_2 = require("./ObservableAPI");
class PromiseBasicApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableBasicApi(configuration, requestFactory, responseProcessor);
    }
    archiveWithHttpInfo(externalEventId, externalAccountId, _options) {
        const result = this.api.archiveWithHttpInfo(externalEventId, externalAccountId, _options);
        return result.toPromise();
    }
    archive(externalEventId, externalAccountId, _options) {
        const result = this.api.archive(externalEventId, externalAccountId, _options);
        return result.toPromise();
    }
    createWithHttpInfo(marketingEventCreateRequestParams, _options) {
        const result = this.api.createWithHttpInfo(marketingEventCreateRequestParams, _options);
        return result.toPromise();
    }
    create(marketingEventCreateRequestParams, _options) {
        const result = this.api.create(marketingEventCreateRequestParams, _options);
        return result.toPromise();
    }
    doCancelWithHttpInfo(externalEventId, externalAccountId, _options) {
        const result = this.api.doCancelWithHttpInfo(externalEventId, externalAccountId, _options);
        return result.toPromise();
    }
    doCancel(externalEventId, externalAccountId, _options) {
        const result = this.api.doCancel(externalEventId, externalAccountId, _options);
        return result.toPromise();
    }
    getByIdWithHttpInfo(externalEventId, externalAccountId, _options) {
        const result = this.api.getByIdWithHttpInfo(externalEventId, externalAccountId, _options);
        return result.toPromise();
    }
    getById(externalEventId, externalAccountId, _options) {
        const result = this.api.getById(externalEventId, externalAccountId, _options);
        return result.toPromise();
    }
    replaceWithHttpInfo(externalEventId, marketingEventCreateRequestParams, _options) {
        const result = this.api.replaceWithHttpInfo(externalEventId, marketingEventCreateRequestParams, _options);
        return result.toPromise();
    }
    replace(externalEventId, marketingEventCreateRequestParams, _options) {
        const result = this.api.replace(externalEventId, marketingEventCreateRequestParams, _options);
        return result.toPromise();
    }
    updateWithHttpInfo(externalEventId, externalAccountId, marketingEventUpdateRequestParams, _options) {
        const result = this.api.updateWithHttpInfo(externalEventId, externalAccountId, marketingEventUpdateRequestParams, _options);
        return result.toPromise();
    }
    update(externalEventId, externalAccountId, marketingEventUpdateRequestParams, _options) {
        const result = this.api.update(externalEventId, externalAccountId, marketingEventUpdateRequestParams, _options);
        return result.toPromise();
    }
}
exports.PromiseBasicApi = PromiseBasicApi;
const ObservableAPI_3 = require("./ObservableAPI");
class PromiseBatchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableBatchApi(configuration, requestFactory, responseProcessor);
    }
    archiveBatchWithHttpInfo(batchInputMarketingEventExternalUniqueIdentifier, _options) {
        const result = this.api.archiveBatchWithHttpInfo(batchInputMarketingEventExternalUniqueIdentifier, _options);
        return result.toPromise();
    }
    archiveBatch(batchInputMarketingEventExternalUniqueIdentifier, _options) {
        const result = this.api.archiveBatch(batchInputMarketingEventExternalUniqueIdentifier, _options);
        return result.toPromise();
    }
    doUpsertWithHttpInfo(batchInputMarketingEventCreateRequestParams, _options) {
        const result = this.api.doUpsertWithHttpInfo(batchInputMarketingEventCreateRequestParams, _options);
        return result.toPromise();
    }
    doUpsert(batchInputMarketingEventCreateRequestParams, _options) {
        const result = this.api.doUpsert(batchInputMarketingEventCreateRequestParams, _options);
        return result.toPromise();
    }
}
exports.PromiseBatchApi = PromiseBatchApi;
const ObservableAPI_4 = require("./ObservableAPI");
class PromiseMarketingEventsExternalApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_4.ObservableMarketingEventsExternalApi(configuration, requestFactory, responseProcessor);
    }
    completeWithHttpInfo(externalEventId, externalAccountId, marketingEventCompleteRequestParams, _options) {
        const result = this.api.completeWithHttpInfo(externalEventId, externalAccountId, marketingEventCompleteRequestParams, _options);
        return result.toPromise();
    }
    complete(externalEventId, externalAccountId, marketingEventCompleteRequestParams, _options) {
        const result = this.api.complete(externalEventId, externalAccountId, marketingEventCompleteRequestParams, _options);
        return result.toPromise();
    }
}
exports.PromiseMarketingEventsExternalApi = PromiseMarketingEventsExternalApi;
const ObservableAPI_5 = require("./ObservableAPI");
class PromiseSearchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_5.ObservableSearchApi(configuration, requestFactory, responseProcessor);
    }
    doSearchWithHttpInfo(q, _options) {
        const result = this.api.doSearchWithHttpInfo(q, _options);
        return result.toPromise();
    }
    doSearch(q, _options) {
        const result = this.api.doSearch(q, _options);
        return result.toPromise();
    }
}
exports.PromiseSearchApi = PromiseSearchApi;
const ObservableAPI_6 = require("./ObservableAPI");
class PromiseSettingsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_6.ObservableSettingsApi(configuration, requestFactory, responseProcessor);
    }
    createWithHttpInfo(appId, eventDetailSettingsUrl, _options) {
        const result = this.api.createWithHttpInfo(appId, eventDetailSettingsUrl, _options);
        return result.toPromise();
    }
    create(appId, eventDetailSettingsUrl, _options) {
        const result = this.api.create(appId, eventDetailSettingsUrl, _options);
        return result.toPromise();
    }
    getAllWithHttpInfo(appId, _options) {
        const result = this.api.getAllWithHttpInfo(appId, _options);
        return result.toPromise();
    }
    getAll(appId, _options) {
        const result = this.api.getAll(appId, _options);
        return result.toPromise();
    }
}
exports.PromiseSettingsApi = PromiseSettingsApi;
const ObservableAPI_7 = require("./ObservableAPI");
class PromiseSubscriberStateChangesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_7.ObservableSubscriberStateChangesApi(configuration, requestFactory, responseProcessor);
    }
    doEmailUpsertByIdWithHttpInfo(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventEmailSubscriber, _options) {
        const result = this.api.doEmailUpsertByIdWithHttpInfo(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventEmailSubscriber, _options);
        return result.toPromise();
    }
    doEmailUpsertById(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventEmailSubscriber, _options) {
        const result = this.api.doEmailUpsertById(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventEmailSubscriber, _options);
        return result.toPromise();
    }
    doUpsertByIdWithHttpInfo(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventSubscriber, _options) {
        const result = this.api.doUpsertByIdWithHttpInfo(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventSubscriber, _options);
        return result.toPromise();
    }
    doUpsertById(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventSubscriber, _options) {
        const result = this.api.doUpsertById(externalEventId, subscriberState, externalAccountId, batchInputMarketingEventSubscriber, _options);
        return result.toPromise();
    }
}
exports.PromiseSubscriberStateChangesApi = PromiseSubscriberStateChangesApi;
