// @hubspot/api-client/lib/codegen/crm/extensions/accounting/models/UpdatedContact.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatedContactCustomerTypeEnum = exports.UpdatedContactSyncActionEnum = exports.UpdatedContact = void 0;
class UpdatedContact {
    static getAttributeTypeMap() {
        return UpdatedContact.attributeTypeMap;
    }
    constructor() {
    }
}
exports.UpdatedContact = UpdatedContact;
UpdatedContact.discriminator = undefined;
UpdatedContact.attributeTypeMap = [
    {
        "name": "syncAction",
        "baseName": "syncAction",
        "type": "UpdatedContactSyncActionEnum",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "emailAddress",
        "baseName": "emailAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "customerType",
        "baseName": "customerType",
        "type": "UpdatedContactCustomerTypeEnum",
        "format": ""
    }
];
var UpdatedContactSyncActionEnum;
(function (UpdatedContactSyncActionEnum) {
    UpdatedContactSyncActionEnum["Create"] = "CREATE";
    UpdatedContactSyncActionEnum["Update"] = "UPDATE";
    UpdatedContactSyncActionEnum["Delete"] = "DELETE";
})(UpdatedContactSyncActionEnum = exports.UpdatedContactSyncActionEnum || (exports.UpdatedContactSyncActionEnum = {}));
var UpdatedContactCustomerTypeEnum;
(function (UpdatedContactCustomerTypeEnum) {
    UpdatedContactCustomerTypeEnum["Contact"] = "CONTACT";
    UpdatedContactCustomerTypeEnum["Company"] = "COMPANY";
})(UpdatedContactCustomerTypeEnum = exports.UpdatedContactCustomerTypeEnum || (exports.UpdatedContactCustomerTypeEnum = {}));
