// @hubspot/api-client/lib/codegen/marketing/forms/models/HubSpotFormDefinitionPatchRequestLegalConsentOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum = exports.HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum = exports.HubSpotFormDefinitionPatchRequestLegalConsentOptions = void 0;
class HubSpotFormDefinitionPatchRequestLegalConsentOptions {
    static getAttributeTypeMap() {
        return HubSpotFormDefinitionPatchRequestLegalConsentOptions.attributeTypeMap;
    }
    constructor() {
    }
}
exports.HubSpotFormDefinitionPatchRequestLegalConsentOptions = HubSpotFormDefinitionPatchRequestLegalConsentOptions;
HubSpotFormDefinitionPatchRequestLegalConsentOptions.discriminator = undefined;
HubSpotFormDefinitionPatchRequestLegalConsentOptions.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum",
        "format": ""
    },
    {
        "name": "subscriptionTypeIds",
        "baseName": "subscriptionTypeIds",
        "type": "Array<number>",
        "format": "int64"
    },
    {
        "name": "lawfulBasis",
        "baseName": "lawfulBasis",
        "type": "HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum",
        "format": ""
    },
    {
        "name": "privacyText",
        "baseName": "privacyText",
        "type": "string",
        "format": ""
    },
    {
        "name": "communicationConsentText",
        "baseName": "communicationConsentText",
        "type": "string",
        "format": ""
    },
    {
        "name": "communicationsCheckboxes",
        "baseName": "communicationsCheckboxes",
        "type": "Array<LegalConsentCheckbox>",
        "format": ""
    },
    {
        "name": "consentToProcessText",
        "baseName": "consentToProcessText",
        "type": "string",
        "format": ""
    },
    {
        "name": "consentToProcessCheckboxLabel",
        "baseName": "consentToProcessCheckboxLabel",
        "type": "string",
        "format": ""
    },
    {
        "name": "consentToProcessFooterText",
        "baseName": "consentToProcessFooterText",
        "type": "string",
        "format": ""
    }
];
var HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum;
(function (HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum) {
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum["None"] = "none";
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum["LegitimateInterest"] = "legitimate_interest";
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum["ExplicitConsentToProcess"] = "explicit_consent_to_process";
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum["ImplicitConsentToProcess"] = "implicit_consent_to_process";
})(HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum = exports.HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum || (exports.HubSpotFormDefinitionPatchRequestLegalConsentOptionsTypeEnum = {}));
var HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum;
(function (HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum) {
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum["Lead"] = "lead";
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum["Client"] = "client";
    HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum["Other"] = "other";
})(HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum = exports.HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum || (exports.HubSpotFormDefinitionPatchRequestLegalConsentOptionsLawfulBasisEnum = {}));
