// @hubspot/api-client/lib/codegen/marketing/forms/models/LegalConsentOptionsLegitimateInterest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalConsentOptionsLegitimateInterestLawfulBasisEnum = exports.LegalConsentOptionsLegitimateInterestTypeEnum = exports.LegalConsentOptionsLegitimateInterest = void 0;
class LegalConsentOptionsLegitimateInterest {
    static getAttributeTypeMap() {
        return LegalConsentOptionsLegitimateInterest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.LegalConsentOptionsLegitimateInterest = LegalConsentOptionsLegitimateInterest;
LegalConsentOptionsLegitimateInterest.discriminator = undefined;
LegalConsentOptionsLegitimateInterest.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalConsentOptionsLegitimateInterestTypeEnum",
        "format": ""
    },
    {
        "name": "subscriptionTypeIds",
        "baseName": "subscriptionTypeIds",
        "type": "Array<number>",
        "format": "int64"
    },
    {
        "name": "lawfulBasis",
        "baseName": "lawfulBasis",
        "type": "LegalConsentOptionsLegitimateInterestLawfulBasisEnum",
        "format": ""
    },
    {
        "name": "privacyText",
        "baseName": "privacyText",
        "type": "string",
        "format": ""
    }
];
var LegalConsentOptionsLegitimateInterestTypeEnum;
(function (LegalConsentOptionsLegitimateInterestTypeEnum) {
    LegalConsentOptionsLegitimateInterestTypeEnum["LegitimateInterest"] = "legitimate_interest";
})(LegalConsentOptionsLegitimateInterestTypeEnum = exports.LegalConsentOptionsLegitimateInterestTypeEnum || (exports.LegalConsentOptionsLegitimateInterestTypeEnum = {}));
var LegalConsentOptionsLegitimateInterestLawfulBasisEnum;
(function (LegalConsentOptionsLegitimateInterestLawfulBasisEnum) {
    LegalConsentOptionsLegitimateInterestLawfulBasisEnum["Lead"] = "lead";
    LegalConsentOptionsLegitimateInterestLawfulBasisEnum["Client"] = "client";
    LegalConsentOptionsLegitimateInterestLawfulBasisEnum["Other"] = "other";
})(LegalConsentOptionsLegitimateInterestLawfulBasisEnum = exports.LegalConsentOptionsLegitimateInterestLawfulBasisEnum || (exports.LegalConsentOptionsLegitimateInterestLawfulBasisEnum = {}));
